app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt

```diff
@@ -74,34 +96,68 @@ import com.github.se.cyrcle.ui.theme.atoms.ScoreStars
 import com.github.se.cyrcle.ui.theme.atoms.SmallFloatingActionButton
 import com.github.se.cyrcle.ui.theme.atoms.Text
 import com.github.se.cyrcle.ui.theme.atoms.ToggleButton
+import com.github.se.cyrcle.ui.theme.defaultOnColor
 import com.github.se.cyrcle.ui.theme.getCheckBoxColors
+import com.github.se.cyrcle.ui.theme.getOutlinedTextFieldColorsSearchBar
 import com.github.se.cyrcle.ui.theme.molecules.BottomNavigationBar
+import com.mapbox.geojson.Point
 import com.mapbox.turf.TurfMeasurement
 import kotlin.math.roundToInt
+import kotlinx.coroutines.runBlocking
 
 const val CARD_HEIGHT = 120
 const val MAX_SWIPE_DISTANCE = 150
+const val maxSuggestionDisplayNameLengthList = 70
 
+@SuppressLint("StateFlowValueCalledInComposition")
 @Composable
 fun SpotListScreen(
     navigationActions: NavigationActions,
     parkingViewModel: ParkingViewModel,
     mapViewModel: MapViewModel,
-    userViewModel: UserViewModel
+    userViewModel: UserViewModel,
+    addressViewModel: AddressViewModel,
+    permissionHandler: PermissionHandler
 ) {
+
   val userPosition by mapViewModel.userPosition.collectAsState()
 
   val filteredParkingSpots by parkingViewModel.closestParkings.collectAsState()
   val pinnedParkings by parkingViewModel.pinnedParkings.collectAsState()
 
-  LaunchedEffect(userPosition) { parkingViewModel.setCircleCenter(userPosition) }
+  // chosen location by the user for the list Screen
+  val chosenLocation: MutableState<Address> = remember {
+    mutableStateOf(
+        Address(
+            latitude = MapConfig.defaultCameraState().center.latitude().toString(),
+            longitude = MapConfig.defaultCameraState().center.longitude().toString()))
+  }
+
+  // value that says wether the user clicked on my Location Suggestion or not
+  val myLocation = remember { mutableStateOf(true) }
+
+  // location permission from location manager
+  val locPermission = permissionHandler.getLocalisationPerm().value
+
+  LaunchedEffect(userPosition, myLocation, chosenLocation.value) {
+    if (locPermission && myLocation.value) parkingViewModel.setCircleCenter(userPosition)
```

Comment:
I like documentation on the ifs, for example the logic of this isn't immediately obvious

Reply:
Well addressed! Thanks!